projects/step-frontend/src/lib/modules/execution/services/execution-tree-paging.ts

```diff
@@ -1,7 +1,43 @@
-import { InjectionToken } from '@angular/core';
+import { inject, Injectable, InjectionToken, OnInit } from '@angular/core';
+import { UserService } from '@exense/step-core';
+import { BehaviorSubject } from 'rxjs';
 
-export type ExecutionTreePaging = Record<string, { skip: number }>;
+export type ExecutionTreePagingSetting = Record<string, { skip: number }>;
 
-export const EXECUTION_TREE_PAGING = new InjectionToken<ExecutionTreePaging>('Execution tree paging');
+export const EXECUTION_TREE_PAGING = new InjectionToken<ExecutionTreePagingSetting>('Execution tree paging');
 
-export const EXECUTION_TREE_PAGE_LIMIT = 1000;
+@Injectable()
+export class ExecutionTreePaging {
+  private _userService = inject(UserService);
+  readonly MAX_NODES_DEFAULT = 1000;
+  private _preferencesFetched: boolean = false;
+  private _customPagingSubject: BehaviorSubject<number>;
+
+  constructor() {
+    this._customPagingSubject = new BehaviorSubject<number>(this.MAX_NODES_DEFAULT);
+  }
+
+  get customPaging$() {
+    return this._customPagingSubject.asObservable();
+  }
```

Comment:
1. We're using `_` prefix for injections only
2. customPagingSubject can be initialized directly in field. Constructor is redundant
3. All observables should have `$` suffix
4. getter accessor is redundant, it a single action, which can be done only once

```suggestion
  private preferencesFetched: boolean = false;
  private customPagingInternal$ = new BehaviorSubject<number>(this.MAX_NODES_DEFAULT);
  readonly customPaging$ = customPagingInternal$.asObservalbe();
```